### ISSUE TITLE
Linking to images on another page on website does not display lightbox

### ISSUE THREAD

**Issue by wavis421** [archetyped/simple-lightbox#879]
## Description of Problem

I'm using an accordion to display recipes, and the images on these recipes display the lightbox as expected.
But if I link to these same recipes from another page and click on the images, the lightbox does not work. Instead I see a full-page image with no way to close.

## Details

* URL of Page exhibiting problem: 
      1) Working (click on Veggie Quesadilla): https://wendywidgets.com/ATasteOfYum/dinner-tortillas-wraps/
      2) Same recipe, not working (click on Week 2, Day 5; then click on the Quesadilla recipe button): 
          https://wendywidgets.com/ATasteOfYum/sample-menus/
* Browser(s) exhibiting problem (Firefox, Chrome, etc.): Chrome, Edge
* Version of SLB Installed: 2.8.1
* Other plugins installed: Beaver Builder lite, Contact Widgets, Insert Headers Footers, Limit Login Attempts, myStickymenu, Ninja Forms, Responsive Accordion, WP Fastest Cache, Yoast SEO.

**Comment by archetyped** [comment]
Hi, SLB appears to be properly displaying images in a lightbox on both pages provided.

You may need to clear your browser's cache to ensure it is not loading an outdated version of the site.  Also make sure that the page has finished loading before clicking on images.

If you are still experiencing this issue after taking the above steps, please provide updated links that exhibit the issue and I would be glad to reopen this ticket and take another look.

Thanks.

**Comment by wavis421** [comment]
I'm using a Chrome incognito window plus cleared the cache. For the link in #2 (https://wendywidgets.com/ATasteOfYum/sample-menus/), the image in the "week 2 day 5" link does work fine. But that's a different image. You then have to click the "Quesadilla" button to get to the same recipe that you see in #1 link. Clicking the image in that recipe is what still does not work.

**Comment by wavis421** [comment]
The path using #2 link to the week2-day5 and then clicking on the Quesadilla recipe button goes to:
https://wendywidgets.com/ATasteOfYum/wp-content/uploads/2021/04/quesadilla-1-scaled.jpg

Using link #1 directly to the Quesadilla recipe, the URL doesn't change when you click the image:
https://wendywidgets.com/ATasteOfYum/dinner-tortillas-wraps/

**Comment by archetyped** [comment]
Are you using a plugin or similar to display the content that is shown when the "Quesadilla" button is clicked?  It looks like that content is loaded after the initial page load, so the image links in that content are not able to be initialized by SLB.

**Comment by wavis421** [comment]
Since these recipes are on a different page, I use JavaScript in the footer to fetch the page where the recipe is located and then copy the innerHTML of the recipe into a <div> on the current page.

Here is my function. The 'recipes' parameter is an array containing the source (from an accordion) and the destination <div> on the current page. Example array entry:   ['ac_1342_collapse3', myQuesadilla]

	function getRecipes(pageName, recipes)
	{
               fetch('https://wendywidgets.com/ATasteOfYum/' + pageName).then(function (response) {
			// The API call was successful!
			return response.text();
		}).then(function (html) {
			// The HTML text string from our response; convert to a document object
			var parser = new DOMParser();
			var doc = parser.parseFromString(html, 'text/html');
				
			// Load each recipe on this page
			for (var i = 0; i < recipes.length; i++) {
				recipes[i][1].innerHTML = doc.getElementById(recipes[i][0]).innerHTML;
			}
				
		}).catch(function (err) {
			// There was an error
			console.warn('Something went wrong.', err);
		});
	}

**Comment by wavis421** [comment]
Forgot to mention what might be an important factor. I use the "Insert Headers and Footers" plug-in to place this code in the footer.

**Comment by archetyped** [comment]
Ah, that explains it.  The HTML is being pulled from another post, but not SLB's metadata for the links in that post, so the image cannot be properly displayed in the lightbox.

If HTML content must be pulled from a different post, then the `SLB.View.assets` array from the other page (which contains necessary metadata for the links on the page), would also need to be retrieved and merged with the `SLB.View.assets` array on the current page.

Once the link metadata is available in the current page's `SLB.View.assets` array, the links in the retrieved HTML should be able to be displayed in the lightbox.

However, note that this is not officially supported and the respective internal code could change in a future update.

**Comment by wavis421** [comment]
I understand the concept of needing to merge the SLB.View.assets on the current page, but how to actually do this is beyond my knowledge base. 

So I decided to leave the menu image and the recipe images all clickable with operational SLB, but images embedded inside the menu not clickable:
     /* nested recipes are not clickable */
     .modal.in .recipe {
	     pointer-events: none;
     }

I then made the menu images into links using the title, so that the user can go to the page where the recipe is located. A compromise, but until I learn how to merge the SLB.View.assets, this will have to do.

Thank-you SO much for your support and help. I hope that in the future, there will be a more automated way to load the SLB links, so I can add it to my JavaScript!!!
